plugins/google.js

```diff
@@ -26,35 +27,132 @@ hoverZoomPlugins.push({
             '$1'
         );
 
-        // Extract direct links to images using document.scripts data (no api call(s) needed)
-        // Mostly inspired from chocolatebot's userscript : Google DWIMages 
+        // Hook Google 'Open' XMLHttpRequests to catch data & metadata associated with pictures displayed
+        // These requests are issued by client to Google servers in order to obtain new data when user scrolls down
+        // Hooked data is stored in scripts in DOM with class='hoverZoomHookedData'
+        if ($('script.hoverZoomHook').length == 0) { // Inject hook script in document if not already there
+            var hookScript = document.createElement('script');
+            hookScript.type = 'text/javascript';
+            hookScript.text = `if (typeof oldXHROpen !== 'function') { // Hook only once!
+                console.log('Hooking XHROpen');
+                oldXHROpen = window.XMLHttpRequest.prototype.open;
+                window.XMLHttpRequest.prototype.open = function(method, url, async, user, password) {
+                    // catch responses
+                    this.addEventListener('load', function() {
+                        try {
+                            //console.log('load: ' + this.responseText);
+                            // store response as plain text in a <script> element for later usage by plug-in
+                            var hookedDataScript = document.createElement('script');
+                            hookedDataScript.type = 'text';
+                            hookedDataScript.text = this.responseText;
+                            hookedDataScript.classList.add('hoverZoomHookedData'); // add specific class for easier retrieval
+                            (document.head || document.documentElement).appendChild(hookedDataScript);
+                            // Add & remove empty <a> element to/from DOM to trigger HoverZoom,
+                            // so data & metadata just added to DOM in <script> element can be exploited
+                            var fakeA = document.createElement('a');
+                            (document.head || document.documentElement).appendChild(fakeA);
+                            (document.head || document.documentElement).removeChild(fakeA);
+                        } catch {}
+                    });
+                    // Proceed with original function
+                    return oldXHROpen.apply(this, arguments);
+                }
+            }`;
+            hookScript.classList.add('hoverZoomHook');
+            (document.head || document.documentElement).appendChild(hookScript);
```

Comment:
I don't understand why do you need to dynamically create a script. All scripts are already executed by the browser, such as for example this `plugins/google.js` file.

Reply:
It's a script with type = 'text', so it'll not be executed by browser.
It's just a place to store chunks of data returned by google when user scrolls down results.

Reply:
Javascript variables can hold state until the page is closed, you don't need to use DOM to store data.

Reply:
I modified this part: XHR responses are now stored in sessionStorage. So no more modifications to DOM needed.